docs/adr/0006-external-types.md

```diff
@@ -0,0 +1,82 @@
+# Replace ViaFfi with FfiConverter to handle external types
+
+* Status: proposed
+* Deciders: bendk, rfkelly, mhammond
+* Consulted: jhugman, jan-erik, travis
+* Date: 2021-08-06
```

Comment:
I'm never sure what to put for "deciders" and "consulted".  I put Ryan and Mark on for deciders since I remember them saying I should make a real PR for the code.  I put James, Jan-Erik, and Travis as consulted since I did some demoing of the code to them yesterday.  Tell me if I should change this and sorry if I forgot anyone's contribution, there's been a lot going on this week!